libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp

```diff
@@ -471,6 +492,180 @@ uint64_t VoteManager::roundDeterminedFromVotes(size_t two_t_plus_one) {
   return 0;
 }
 
+void VoteManager::sendRewardPeriodCertVotes(uint64_t reward_period) {
+  auto reward_period_cert_votes = db_->getCertVotes(reward_period);
+  if (reward_period_cert_votes.empty()) {
+    return;
+  }
+
+  auto net = network_.lock();
+  assert(net);
+  net->onNewPbftVotes(std::move(reward_period_cert_votes), true);
+}
+
+bool VoteManager::addRewardVote(const std::shared_ptr<Vote>& vote) {
+  // Don't check the vote if in DB here. That will check in updateRewardVotes for reducing DB read
+  const auto vote_hash = vote->getHash();
+  std::unique_lock lock(reward_votes_mutex_);
+  if (reward_votes_.second.contains(vote_hash)) {
+    return false;
+  }
+
+  const auto voted_block_hash = vote->getBlockHash();
+  const auto pbft_chain_last_block_hash = pbft_chain_->getLastPbftBlockHash();
+  if (voted_block_hash != pbft_chain_last_block_hash) {
```

Comment:
so what if some node is 1 period ahead of you. You are at period 10 and the other node is at period 11 - `pbft_chain_->getLastPbftBlockHash()` for you here returns different hash than `vote->getBlockHash()`. If the node starts to send you rewards votes, you will drop them all because this condition will fail ?

Reply:
Good point.
we can remove seen votes here https://github.com/Taraxa-project/taraxa-node/blob/reward_cert_votes/libraries/core_libs/network/src/tarcap/packets_handlers/vote_packet_handler.cpp#L31 and that will be able to get it again.

I know that's bad. but I don't have any better solutions so far. Do you have any ideas?